### DISCUSSION TITLE
Feature Requests & Progress

### DISCUSSION THREAD

**Discussion by BigJk** [BigJk/snd#16] [Ideas]
If you are missing a feature in Sales & Dungeons and it's not already on the todo in the bottom feel free to suggest it! If it sounds reasonable I will add it to the todo 😄 If a item is checked but the feature is not present in the current version it will probably arrive in the next release.

If you found a bug please create a issue: https://github.com/BigJk/snd/issues

## Template Page

- [ ] Some basic default templates that user can duplicate and tinker with (In Progress: https://github.com/BigJk/snd-default-templates)
- [ ] Caching on overview page to avoid long loading times.
- [x] Different view modes
- [x] Global settings for templates that are available in all entries and can be configured

## Generator Page

- [ ] Caching on overview page to avoid long loading times.
- [x] Screenshot to .PNG function like on a template
- [x] Different view modes
- [x] Save different user configs for a generator by name

## Workshop

- [x] Add a "Import All" option / button to import everything a user has published
- [x] Add Search / Filter to better browse packages
- [x] Notify and / or download missing data sources of a imported package
- [ ] Show if the latest version of a package is installed or if update is available

## Data Sources

- [x] Edit data source entries in S&D
- [ ] Raw JSON import

## Settings

- [ ] Add "test print" button

## Other

- [ ] Batch export all templates, generators and data sources

## Wiki / Documentation

- [x] Create a FAQ
- [x] Create a page describing data source format

**Comment by frshbeetz** [comment]
The ability to edit existing entries in a data source in the application would be cool.

**Comment by frshbeetz** [comment]
The ability to export one's entire environment, "export all DSs/Gens/Templates/ETC" would be nice QOL for updating software versions.

**Reply by BigJk** [reply]
Good idea. In general you can just overwrite the files of your version with the newest. All data is stored in the `userdata` folder which won't be overwritten 😄

**Comment by frshbeetz** [comment]
The Generators page gets heavy, since it has to render every generator's preview. 
* Ability to disable the previews on the generators page. 
 Or
* Perhaps it should render a static "preview" image when one "saves" the generator.

(same applies for the Templates page really)

**Reply by BigJk** [reply]
Good Idea! I will think about a way to cache them efficiently to avoid re-render.

**Comment by Naeit** [comment]
Is it possible to export a .PNG instead of having to print it out?

**Reply by BigJk** [reply]
If you open a Template you can select a Entry and there should be a screenshot button on the selected Entry. That will create a PNG. For generators it's missing atm. Does that help?

**Reply by Naeit** [reply]
Yes thank you.
I will try it out.

I don't have a Thermal Printer but I think it is an awesome tool.
I want to use it anyways.
It looks practical and I can always get a printer later 🙂

**Comment by tgerardot** [comment]
A way to bulk import a large amount of json entries when editing a data source would be very helpful. 

Also, is the discord link not functioning correctly? I'm unable to join the server.

**Reply by BigJk** [reply]
Are you using this link? https://discord.com/invite/5MUZEjc

Seems working for me 🤔

**Reply by tgerardot** [reply]
Hmmm, yes. Here's what I'm seeing. I've tried on multiple devices. Never seen it before when joining a Discord.
![discord-whoops](https://user-images.githubusercontent.com/33664788/220993889-f415e604-3579-40b1-80b4-a7a1ff8b7bf3.png)

**Reply by BigJk** [reply]
Strange. It's a Community Server so you need to have a verified email. There are people joining every day via the link so I'm not sure what the problem might be 🤔

**Reply by tgerardot** [reply]
Solved. Oddly, couldn't just follow the link. Had to directly add to discord. Thank you for your patience with me
